src/main/java/com/untamedears/JukeAlert/listener/JukeAlertListener.java

```diff
@@ -204,74 +208,76 @@ public void placeSnitchBlock(BlockPlaceEvent event) {
 			return;
 		}
 		Block block = event.getBlock();
-		if (block.getType().equals(Material.JUKEBOX)) {
-			Player player = event.getPlayer();
-			Location loc = block.getLocation();
-			Reinforcement rei = event.getReinforcement();
-			if (rei instanceof PlayerReinforcement) {
-				PlayerReinforcement reinforcement = (PlayerReinforcement) rei;
-				Group owner = reinforcement.getGroup();
-				if (owner == null) {
-					JukeAlert.getInstance().log(String.format(
-							"No group on rein (%s)", reinforcement.getLocation().toString()));
-				}
-				Snitch snitch;
-				if (snitchManager.getSnitch(loc.getWorld(), loc) != null) {
-					snitch = snitchManager.getSnitch(loc.getWorld(), loc);
-					plugin.getJaLogger().updateSnitchGroup(
-						snitchManager.getSnitch(loc.getWorld(), loc), owner.getName());
-					snitchManager.removeSnitch(snitch);
-					snitch.setGroup(owner);
-				} else {
-					snitch = new Snitch(loc, owner, true, false);
-					plugin.getJaLogger().logSnitchPlace(
-						player.getWorld().getName(), owner.getName(), "", loc.getBlockX(), loc.getBlockY(),
-						loc.getBlockZ(), true);
-					snitch.setId(plugin.getJaLogger().getLastSnitchID());
-					plugin.getJaLogger().increaseLastSnitchID();
-				}
-				snitchManager.addSnitch(snitch);
-
-				player.sendMessage(
-					ChatColor.AQUA + "You've created a snitch block registered to the group " + owner.getName()
-					+ ".  To name your snitch, type /janame.");
-			}
-		} else if (block.getType().equals(Material.NOTE_BLOCK)) {
-			Player player = event.getPlayer();
-			Location loc = block.getLocation();
-			Reinforcement rei = event.getReinforcement();
-			if (rei instanceof PlayerReinforcement) {
-				PlayerReinforcement reinforcement = (PlayerReinforcement) rei;
-				Group owner = reinforcement.getGroup();
-				if (owner == null) {
-					JukeAlert.getInstance().log(String.format(
-							"No group on rein (%s)", reinforcement.getLocation().toString()));
-				}
-				Snitch snitch;
-				if (snitchManager.getSnitch(loc.getWorld(), loc) != null) {
-					snitch = snitchManager.getSnitch(loc.getWorld(), loc);
-					plugin.getJaLogger().updateSnitchGroup(
-						snitchManager.getSnitch(loc.getWorld(), loc), owner.getName());
-					snitchManager.removeSnitch(snitch);
-					snitch.setGroup(owner);
-				} else {
-					snitch = new Snitch(loc, owner, false, false);
-					plugin.getJaLogger().logSnitchPlace(
-						player.getWorld().getName(), owner.getName(), "", loc.getBlockX(), loc.getBlockY(),
-						loc.getBlockZ(), false);
-					snitch.setId(plugin.getJaLogger().getLastSnitchID());
-					plugin.getJaLogger().increaseLastSnitchID();
-				}
-				snitchManager.addSnitch(snitch);
-
-				player.sendMessage(ChatColor.AQUA + "You've created an entry snitch registered to the group "
-					+ owner.getName() + ".  To name your entry snitch, type /janame.");
-			}
+		if (block == null || block.getType() == null) {
+			return;
+		}
+		boolean isJukebox = block.getType().equals(Material.JUKEBOX);
```

Comment:
You need to check for isJukebox and isNoteblock, and fast-fail if neither.

As it is, this'll try to create snitches for every kind of block you reinforce.